### PR TITLE
Skip backend integration tests when cli flag isn't passed

### PR DESCRIPTION
**Pull request by yebrahim**
Backend unit tests now run by default if you run `go test ./...`. If you want integration tests to run as well, set the `unitTestsOnly` flag to false: `$ go test ./... -args -unitTestsOnly=false`.

/area back-end
/assign @neuromage

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/527)

### CODE REVIEW

File: test/backend-unit-test/run_test.sh [hunk 1]
```diff
@@ -51,7 +51,7 @@ cd "${BASE_DIR}/${TEST_DIR}"
 
 # Run test and store the exit code.
 echo "Run unit test..."
-TEST_RESULT=`go test -v ./... 2>&1`
+TEST_RESULT=`go test -v ./... -args -unitTestsOnly=false 2>&1`
```
Comment (Ark-kun) [inline test/backend-unit-test/run_test.sh]: Can we somehow remove the double negation? E.g. `runIntegrationTests=true`

Comment (yebrahim) [reply test/backend-unit-test/run_test.sh]: Done.

**Comment by yebrahim** [comment]
@IronPan @neuromage does it make sense to run backend unit tests in Travis rather than Prow? Why wait for 17+ minutes to find out the result of these tests when you can wait for 3-4 minutes?

**Comment by vicaire** [comment]
I am in favor or running the unit tests in travis as well if it's faster.

(Maybe we need to run them in prow as well to get the test grid? Yang can
comment on that part).

**Comment by IronPan** [comment]
the unit tests are not taking long to finish and show up the result in github. but feel free to move them to travis if you want.

**Comment by yebrahim** [comment]
@IronPan this needs a /approve as well.
When I looked, it was taking a long time for the unit tests step to run, it seems like it waits for the build image step?

**Comment by IronPan** [comment]
The unit tests and build image are parallel. UT won't depend on the images being built

**Comment by yebrahim** [comment]
Need one more lgtm here after resolving conflicts with master.

**Pull request merged**